modules/channel-vonage/src/backend/client.ts

```diff
@@ -109,23 +111,58 @@ export class VonageClient {
 
     const conversation = await this.conversations.recent(userId)
 
+    const messageContent = <ExtendedChannelContent>body.message.content
+
     // https://developer.nexmo.com/api/messages-olympus?theme=dark
     let payload: sdk.IO.IncomingEvent['payload'] = null
-    switch (body.message.content.type) {
+    switch (messageContent.type) {
       case 'text':
-        const text = body.message.content.text
+        const text = messageContent.text
         payload = { type: 'text', text }
 
         // Since single choice and carousel are down rendered, we have to handle option selection differently
         payload = (await this.handleOptionResponse(text, userId, conversation.id)) ?? payload
         break
       case 'audio':
-        // TODO: Link received from Nexmo/Vonage API are only valid for 10min
-        const audio = body.message.content.audio.url
-        payload = { type: 'audio', audio }
+        payload = {
+          type: 'voice',
+          url: messageContent.audio.url
+        }
+        break
+      case 'image':
+        payload = {
+          type: 'file', // TODO: Change image content-type type to 'image' instead of 'file'
+          title: messageContent.image.caption,
+          url: messageContent.image.url,
+          collectFeedback: false
+        }
+        break
+      case 'video':
+        payload = {
+          type: messageContent.type,
+          title: `${path.basename(messageContent.video.url)}.mp4`, // TODO: Take for granted that video file will be .mp4 files?
+          url: messageContent.video.url,
+          collectFeedback: false
+        }
+        break
+      case 'file':
+        payload = {
+          type: 'document',
+          title: messageContent.file.caption,
+          url: messageContent.file.url,
+          collectFeedback: false
+        }
+        break
```

Comment:
So I can see here that you put all the links that `storeFileLocally` needs into the `url` property. This makes for a simpler action but it only works because, at the moment, content renderers take the image, video and audio data and converts them into standard payloads (that all use the `url` property). 

When we convert to channel renderers, we won't be putting everything into the `url` property anymore (audio uses `.audio`, video `.video`, image `.image`). So either we migrate all "media" content types to use the same property in the jsonSchema or we do `.audio || .video || .image` in the `storeFileLocally` logic when looking for a url

Reply:
Personally, I think it would be better to have a `url` property instead of `.audio || .video || .image`. I find it kind of weird to have both a `type` property that specifies what kind of payload it is and a specific property for each of them to store a URL.